pkg/systemd/overview-cards/realmd.jsx

```diff
@@ -67,10 +67,13 @@ export class RealmdClient {
     onClose(ev, options) {
         if (options.problem === "not-found") {
             // see if we can install it
-            packagekit.detect().then(exists => {
-                if (exists) {
+            packagekit.detect_with_details().then(details => {
+                if (details.available) {
                     this.error = _("Joining a domain requires installation of realmd");
                     this.install_realmd = true;
+                } else if (details.reason == "immutable-os") {
+                    this.error = _("Joining a domain requires adding of realmd to the OS image");
+                    this.install_realmd = true;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18240-20230127-135755-f796be5b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)